rmw_iceoryx_cpp/src/internal/iceoryx_type_info_introspection.cpp

```diff
@@ -218,6 +313,23 @@ std::string iceoryx_get_message_name(const rosidl_message_type_support_t * type_
   return "";
 }
 
+std::string iceoryx_get_service_name(const rosidl_service_type_support_t * type_supports)
+{
+  auto ts = get_type_support(type_supports);
+
+  if (ts.first == TypeSupportLanguage::CPP) {
+    auto members =
+      static_cast<const rosidl_typesupport_introspection_cpp::ServiceMembers *>(ts.second->data);
+    return members->service_name_;
+  } else if (ts.first == TypeSupportLanguage::C) {
+    auto members =
+      static_cast<const rosidl_typesupport_introspection_c__ServiceMembers *>(ts.second->data);
+    return members->service_name_;
+  }
+  // Something went wrong
+  return "";
```

Comment:
@elBoberido I understand what you mean. However, `std::unreachable` would be wrong as there might be other language type supports than C or C++. Maybe the comment is misleading here. Any error with the extraction of the type support will be reported by `get_{service,message}_typesupport_handle`. If you don't mind I would keep this unchanged for now and tackled it with a wider refactoring.

Reply:
Sure. It's up to you. This just comes from the philosophy to fail fast instead of hunting down a bug :)

The same pattern was also used at different places so it should be done in a consistent way and if changed, it should therefore be changed everywhere which would blow up this PR.